### PR TITLE
Implement Payload File name with date for LA-PHL

### PR DESCRIPTION
**Pull request by oslynn**
This PR implements the Payload File name with a date for LA-PHL.  It used Report.formFileName utility to generate the filename to send to receiver.  The utility generate filename as: <organizationName>-<reported>-reportCreatedAt.HL7 or .CSV (according to the receiver.translation.type).

Test Steps:
1. ./gradlew clean package
2. docker-compose up
3. ./gradlew :testIntegration --tests "gov.cdc.prime.router.transport.RESTTransportIntegrationTests.test  transport postReport with valid file name for Natus"

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue
https://app.zenhub.com/workspaces/engagement-6166edbd409257001e09f1b2/issues/gh/cdcgov/prime-reportstream/12719

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

**Comment by adegolier** [comment]
Is it okay to use colons in a filename? That seems... weird.... but if that's fine, lgtm

File: prime-router/src/main/kotlin/transport/RESTTransport.kt [hunk 1]
```diff
@@ -90,7 +91,25 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
         val receiver = header.receiver ?: error("No receiver defined for report $reportId")
         val reportContent: ByteArray = header.content ?: error("No content for report $reportId")
         // get the file name, or create one from the report ID, NY requires a file name in the POST
-        val fileName = header.reportFile.externalName ?: "$reportId.hl7"
+        val fileName = if (restTransportInfo.headers["File-Name"] != null) {
```
Comment (victor-chaparro) [inline prime-router/src/main/kotlin/transport/RESTTransport.kt:94]: Why can we not use the existing function inside Report.kt:formFilename to build the filename that uses the receiver setting fileFormat and the file-name-templates.yml?

Comment (oslynn) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt]: Thank you, I used the utility to generate the fileName.

File: prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt [hunk 2]
```diff
@@ -635,6 +636,56 @@ hnm8COa8Kr+bnTqzScpQuOfujHcFEtfcYUGfSS6HusxidwXx+lYi1A==
...
     }
 
+    @Test
+    fun `test  transport postReport with valid file name for Natus`() {
+        val header = makeHeader()
+        val mockRestTransport = spyk(RESTTransport(mockClientStringTokenOk()))
+
+        // Given:
+        //      lookupDefaultCredential returns mock UserPassCredential object to allow
+        //      the getAuthTokenWithUserPass() to be called.
```
Comment (victor-chaparro) [inline prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt:646]: Why are you copying the filename logic here? You shouldn't have to copy the same logic you're testing in the unit test. You should be calling the function that has this logic instead. If  you make changes to the function you won't have unit tests that cover that.

**Review by victor-chaparro — CHANGES_REQUESTED** [review]
Unit test needs to call function not re-create the logic inside.

File: prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt (hunk 2 above)
Comment (oslynn) [reply prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt:647]: Thank you.

**Review by victor-chaparro — APPROVED** [review]
Looks good!

**Pull request merged**